phoebe/frontend/bundle.py

```diff
@@ -11912,8 +11933,10 @@ def restore_conf():
                 # TODO: have this return a dictionary like pblums/l3s that we can pass on to the backend?
 
                 # we need to check both for enabled but also passed via dataset kwarg
-                ds_kinds_enabled = self.filter(dataset=dataset_this_compute, context='dataset', **_skip_filter_checks).kinds
-                if 'lc' in ds_kinds_enabled or 'rv' in ds_kinds_enabled or 'lp' in ds_kinds_enabled:
+                # ds_kinds_enabled = self.filter(dataset=dataset_this_compute, context='dataset', **_skip_filter_checks).kinds
+                datasets = self._datasets_that_require_meshing(compute)
```

Comment:
looking at this closer, `compute_l3s` defaults to a slightly different list of datasets (those with `l3_mode` parameters).  Right now these are just `lc` datasets which also are always including in `_datasets_that_require_meshing`, but I'm a bit worried this could break in the future.  We could:
* add a comment that hopefully we'll see if we ever have any other dataset type use third light (or ever allow light curves without needing meshing, but still needing l3 logic)
* do something like `set(self._datasets_that_require_meshing(compute) + self._datasets_that_require_l3(compute))` 😬

Reply:
I reworked the logic, see what you think in the commit that's about to hit the branch.